### PR TITLE
Update click tooltip styles to match design library

### LINKED ISSUE #6917: Update the Balances/Deposits clickable tooltips in Overview to the correct component
### Describe the bug
We need to update the black tooltips that is currently implemented in WooCommerce Payments Overview (and possibly elsewhere within WCPay?) to the correct clickable white tooltips for better accessibility/readability and align with the new Woo design direction. 

We have 2 tooltip components available for Woo:
1. black tooltip, activated on hover, ideal for shorter notices
2. white clickable tooltips (popovers), activated on click, ideal for longer form messages

### To Reproduce

1. Go to Woo Commerce Payments
2. Click on a "clickable" icon or pill

### Actual behavior
Clicking a help icon, or any clickable element that triggers a tooltip shows the black tooltip.

### Screenshots
image (3)
image (1)

### Expected behavior
We need to use the clickable white tooltip component instead, found [in the design system Figma](https://www.figma.com/file/Kofpo8jj4zDWXDsVdlSz0f/WooCommerce-Admin-Design-Library?type=design&node-id=6795%3A10109&mode=design&t=JMWLeP4FlzoDURdH-1). If it's clickable, it should trigger the white popover, not the black tooltip. We can keep the same positioning of the popover that is configured 
The component doesn't seem to be in Storybook, but this was first used for the new Woo product form

image

### PR DESCRIPTION
**Pull request by Jinksi**
Fixes #6917

#### Changes proposed in this Pull Request

This PR updates the style of the [`ClickTooltip`](https://github.com/Automattic/woocommerce-payments/blob/35b5b303adfbac06fa2be5a7e6c2635102a6c6de/client/components/tooltip/index.tsx#L79) tooltip element to match the design of the Kofpo8jj4zDWXDsVdlSz0f-fi-6795%3A10109.

tooltip design

**Notes**
* the ClickTooltip component is currently only used for the Account Balances Tooltip Panel ([98c2911/client/components/account-balances/index.tsx](https://github.com/Automattic/woocommerce-payments/blob/98c291141b41f87213f480a4ce271260fef29fb8/client/components/account-balances/index.tsx)), but will be used as part of the new Transaction/Dispute details screen being introduced in #6883.
* For `ClickTooltip`, I've removed the arrow that appears for `HoverTooltip` to match the design.
* I've refactored `HoverTooltip` and `ClickTooltip` to accept an optional `buttonIcon` prop, similar to [BannerNotice](https://github.com/Automattic/woocommerce-payments/blob/e6f547d53410febbc11416656afc7e72a571e1ea/client/components/banner-notice/index.tsx). This allows us to keep the color/tab/accessibility logic abstracted within the tooltip component.
* The tooltip button icon and any tooltip content links will be coloured as the _WP admin theme color_ or _Gutenberg Blue_

#### Testing instructions

* View the **Payments → Overview** screen in WP-admin.
* Click on the help icon next to `Available funds` or `Pending funds` to view the tooltip content.
* Ensure the tooltip content matches the design with a white background.

**Before**
image

**After**
image

https://github.com/Automattic/woocommerce-payments/assets/3147296/572774ad-ccb8-4561-9b4a-30ee7d68298e

**Using a custom WP admin theme color (WooExpress)**
image

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️) Changes to CSS styles – testing not suitable as per current testing strategy
- [x] Tested on mobile – tested via Safari on iOS 16.6

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6999` or branch name `fix/6917-click-tooltip-styles` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/6917-click-tooltip-styles) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: a4bbb6b751b4d568a9ca5ee3e7661c11f1611b26
- Build time: 2023-08-22 06:38:30 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Comment by Jinksi** [comment]
@nikkivias I've tagged you as reviewer to get your feedback – this branch is currently running on the [Helix Demo site](https://woo-delectably-sparkly-ninja.wpcomstaging.com/wp-admin/admin.php?page=wc-admin&path=/payments/overview). Let me know what you think 🙏

**Comment by nikivias** [comment]
### My review
Hi team so I noticed that by default the help icons are colored. And it's also triggering a weird black box around the help icon when clicked. Sidenote: That red dispute warning icon is oddly huge. @Jinksi 


https://github.com/Automattic/woocommerce-payments/assets/99638528/f03861b3-2a75-41b5-b0bd-a4e9f9448937



### What we want instead
Use a grey color by default, color it on hover, open tooltip on click. [I mocked it up on Figma](https://www.figma.com/file/q2b4LTEUhyw0y1vKmfEVpx/%5BWCPay%5D-Deposits-UI-Improvements?type=design&node-id=2916%3A20216&mode=design&t=0rrdy4gNZBGrtC8u-1) 


https://github.com/Automattic/woocommerce-payments/assets/99638528/916d7e4f-af8f-496a-beda-9d8731e84ef2

**Review by shendy-a8c — COMMENTED** [review]
Code diff looks good and tested well. I now see the popover with white background.

Will review again after @nikkivias' feedback is addressed.

**Comment by Jinksi** [comment]
Thanks @nikkivias !

I've made further changes. I hadn't included any changes to the tooltip button colours previously. Thanks for mocking this up in Figma.

The button now will be Gutenberg Grey 600 unless hovered or in focus (using keyboard tab navigation) when it will become Gutenberg Blue OR the WC admin theme color.

https://github.com/Automattic/woocommerce-payments/assets/3147296/572774ad-ccb8-4561-9b4a-30ee7d68298e

> That red dispute warning icon is oddly huge
This should match the design of the Payments Overview task list item. RiEQmaKRI7u9USAI3lyZz0VX-fi-2885_183722

@shendy-a8c I've refactored the tooltip icon to now be an optional prop, similar to [BannerNotice](https://github.com/Automattic/woocommerce-payments/blob/e6f547d53410febbc11416656afc7e72a571e1ea/client/components/banner-notice/index.tsx). This allows us to keep the color/tab/accessibility logic abstracted within the tooltip component.

File: client/components/account-balances/style.scss [hunk 1]
```diff
@@ -31,21 +31,6 @@
 	align-items: center;
 }
 
-.wcpay-account-balances__balances__item__tooltip-button {
```
Comment (Jinksi) [inline client/components/account-balances/style.scss:34]: Styling for tooltip button is now encapsulated in Tooltip component styles.

File: client/components/account-balances/balance-tooltip.tsx [hunk 2]
```diff
@@ -22,19 +22,9 @@ const BalanceTooltip: React.FC< BalanceTooltipProps > = ( {
...
-				tabIndex={ 0 }
-			>
-				<HelpOutlineIcon
-					size={ 18 }
-					className="wcpay-account-balances__balances__item__tooltip-button__icon"
-				/>
-			</div>
-		</ClickTooltip>
+			buttonIcon={ <HelpOutlineIcon /> }
+			buttonLabel={ label }
```
Comment (Jinksi) [inline client/components/account-balances/balance-tooltip.tsx:26]: Moving styling and required markup/attrs to the Tooltip component now simplifies API and makes it easier to use for the common use-case of using an icon as the tooltip button.

**Review by shendy-a8c — APPROVED** [review]
Looks good!

**Comment by Jinksi** [comment]
Do you want to take another look at [the demo site](https://woo-delectably-sparkly-ninja.wpcomstaging.com/wp-admin/admin.php?page=wc-admin&path=/payments/overview) @nikkivias or are you satisfied with my changes in the screen recording above https://github.com/Automattic/woocommerce-payments/pull/6999#issuecomment-1683366162?

**Comment by nikivias** [comment]
Hey @Jinksi really sorry for having missed this. Looks goooood!

**Comment by Jinksi** [comment]
No problem at all, thanks @nikkivias! 🙏

**Pull request merged**